### PR TITLE
Bump clangsharp to 15.0.3

### PR DESCRIPTION
**Pull request by Beyley**
# Summary of the PR
Updates the ClangSharp used by BuildTools to the latest 15.0.3, rather than an old build under Ultz

# Related issues, Discord discussions, or proposals
https://discord.com/channels/521092042781229087/587346162802229298/1041085075766386778

# Further Comments
still wip

### CODE REVIEW

**Comment by Perksey** [comment]
@Beyley can you fix the D3D11 example project? It seems to have been bitten by the in->ref change. Looks good other than that

**Comment by Beyley** [comment]
CI passing @Perksey

File: examples/CSharp/Direct3D11 Tutorials/Tutorial 1.2 - Hello quad/Program.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ ref swapchain
             PSysMem = vertexData
         };
 
-        SilkMarshal.ThrowHResult(device.CreateBuffer(ref bufferDesc, ref subresourceData, ref vertexBuffer));
```
Comment (Perksey) [inline examples/CSharp/Direct3D11 Tutorials/Tutorial 1.2 - Hello quad/Program.cs:182]: We should probably discourage use of implicit in as this makes an implicit copy and takes the reference to that instead, whereas we can use `in bufferDesc` etc

Comment (Beyley) [reply examples/CSharp/Direct3D11 Tutorials/Tutorial 1.2 - Hello quad/Program.cs:182]: Ĉi tio estas Vera DotNet Momento
ill add the explicit `in` rn

**Comment by Perksey** [comment]
Where did this disappear to????

**Comment by Beyley** [comment]
no idea

**Pull request merged**